DQMOffline/Trigger/plugins/RazorMonitor.cc

```diff
@@ -0,0 +1,421 @@
+  setMETitle(MRVsRsq_ME_,"M_{R} [GeV]","R^{2}");
+
+  // 2D hist, MR & Rsq, with tight jet requirement
+  histname = "MRVsRsq_Tight"; histtitle = "PF MR vs PF Rsq";
+  bookME(ibooker,MRVsRsq_Tight_ME_,histname,histtitle,mr_binning_, rsq_binning_);
+  setMETitle(MRVsRsq_Tight_ME_,"M_{R} [GeV]","R^{2}");
+
+  // Initialize the GenericTriggerEventFlag
+  if ( num_genTriggerEventFlag_ && num_genTriggerEventFlag_->on() ) num_genTriggerEventFlag_->initRun( iRun, iSetup );
+  if ( den_genTriggerEventFlag_ && den_genTriggerEventFlag_->on() ) den_genTriggerEventFlag_->initRun( iRun, iSetup );
+
+}
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+void RazorMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup)  {
+
+  // Filter out events if Trigger Filtering is requested
+  if (den_genTriggerEventFlag_->on() && ! den_genTriggerEventFlag_->accept( iEvent, iSetup) ) return;
+
+  //met collection
+  edm::Handle<reco::PFMETCollection> metHandle;
+  iEvent.getByToken( metToken_, metHandle );
+  reco::PFMET pfmet = metHandle->front();
+  if ( ! metSelection_( pfmet ) ) return;
+
+  //jet collection, track # of jets for two working points
+  edm::Handle<reco::PFJetCollection> jetHandle;
+  iEvent.getByToken( jetToken_, jetHandle );
+  std::vector<reco::PFJet> jets;
+  std::vector<reco::PFJet> tightJets;
+  if ( int(jetHandle->size()) < njets_ ) return;
+  for ( auto const & j : *jetHandle ) {
+    if ( jetSelection_( j ) ) jets.push_back(j);
+    if ( tightJetSelection_( j ) ) tightJets.push_back(j);
+  }
+  if ( int(jets.size()) < njets_ ) return;
+
+  //razor hemisphere clustering from previous step
+  edm::Handle< vector<math::XYZTLorentzVector> > hemispheres;
+  iEvent.getByToken (theHemispheres_,hemispheres);
+
+  if (not hemispheres.isValid()){
+    return;
+  }
+
+  if(hemispheres->size() ==0){  // the Hemisphere Maker will produce an empty collection of hemispheres if # of jets is too high
+    edm::LogError("DQM_HLT_Razor") << "Cannot calculate M_R and R^2 because there are too many jets! (trigger passed automatically without forming the hemispheres)" << endl;
+    return;
+  }
+
+  if(hemispheres->size() != 0 && hemispheres->size() != 2 && hemispheres->size() != 5 && hemispheres->size() != 10){
+    edm::LogError("DQM_HLT_Razor") << "Invalid hemisphere collection!  hemispheres->size() = " << hemispheres->size() << endl;
+    return;
+  }
+
+  //define hemispheres
+  TLorentzVector ja(hemispheres->at(0).x(),hemispheres->at(0).y(),hemispheres->at(0).z(),hemispheres->at(0).t());
+  TLorentzVector jb(hemispheres->at(1).x(),hemispheres->at(1).y(),hemispheres->at(1).z(),hemispheres->at(1).t());
+
+  //dummy vector (this trigger does not care about muons)
+  std::vector<math::XYZTLorentzVector> muonVec;
```

Comment:
We don't have any triggers that care about muons, no. Can you elaborate on what you mean by performance of other paths?